### PR TITLE
Implement tracer provider in SDK

### LINKED ISSUE #282: SDK: Implement the spi.TracerProvider
Allow users to configure different parts of the TracerSdk via manifest files.

### PR DESCRIPTION
**Pull request by pavolloffay**
Resolves #282 

Signed-off-by: Pavol Loffay <ploffay@redhat.com>

### CODE REVIEW

**Comment by pavolloffay** [comment]
I will merge this but I am happy to fix comments after the merge.

**Comment by pavolloffay** [comment]
It seems that I cannot merge PRs

**Comment by carlosalberto** [comment]
@pavolloffay You should now ;)

File: sdk/src/test/java/io/opentelemetry/sdk/trace/TracerSdkProviderTest.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+package io.opentelemetry.sdk.trace;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import org.junit.Test;
+
+public class TracerSdkProviderTest {
+
+  @Test
+  public void testDefault() {
```
Comment (bogdandrutu) [inline sdk/src/test/java/io/opentelemetry/sdk/trace/TracerSdkProviderTest.java]: What about a test that OpenTelemetry.getTracer returns an instance of the TracerSdk?

Comment (pavolloffay) [reply sdk/src/test/java/io/opentelemetry/sdk/trace/TracerSdkProviderTest.java]: Could you please clarify what are you asking? 

`OpenTelemetryTest` should test it:

```java
  @Test
  public void testDefault() {
    assertThat(OpenTelemetry.getTracer()).isInstanceOf(TracerSdk.class);
    assertThat(OpenTelemetry.getMeter()).isInstanceOf(DefaultMeter.getInstance().getClass());
    assertThat(OpenTelemetry.getTagger()).isInstanceOf(DefaultTagger.getInstance().getClass());
  }
```

**Comment by pavolloffay** [comment]
@carlosalberto I do not have rights to merge anything in this repo

File: sdk/src/test/java/io/opentelemetry/sdk/trace/TracerSdkProviderTest.java (hunk 1 above)
Comment (bogdandrutu) [reply sdk/src/test/java/io/opentelemetry/sdk/trace/TracerSdkProviderTest.java]: @Test
  public void loadTraceSdk() {
    assertThat(OpenTelemetry.getTracer()).isInstanceOf(TracerSdk.class);
  }

Comment (bogdandrutu) [reply sdk/src/test/java/io/opentelemetry/sdk/trace/TracerSdkProviderTest.java]: I see you said in the OpenTelemetryTest. That is also fine.

**Comment by bogdandrutu** [comment]
Currently is enforced that only maintainers can merge PRs, we need to talk about this in the board committee and clarify who has the right to do it.

**Pull request merged**